content/techniques/performance.md

```diff
@@ -74,6 +74,38 @@ You can pass options into the Fastify constructor through the `FastifyAdapter` c
 new FastifyAdapter({ logger: true });
 ```
 
+
+#### Middleware
+
+The middleware with `fastify` "works", but it retrieves only the raw `req` and `res` objects and not the fastify wrapper. This is a side effect of how `middie` works and `fastify` already talks about it in its documents for more information see [here](https://www.fastify.io/docs/latest/Reference/Middleware/).
+
+> warning **Warning** The middleware in `fastify` has an IncomingMessage request type `FastifyRequest['raw']` and a ServerResponse type `FastifyReply['raw']`.
```

Comment:
```suggestion
> warning **Warning** The middleware in `fastify` has an IncomingMessage request type `FastifyRequest['raw']` and a ServerResponse type `FastifyReply['raw']`.
```

cc @jmcdo29 any ideas on how to rewrite it to make it sound more native-ish 🤔

Reply:
```suggestion
> warning **Warning** When using `fastify` as the HTTP engine the request parameter of middleware has a type of `FastifyRequest['raw']` and the response parameter has a type `FastifyReply['raw']`.
```